File: 백준/Gold/5052. 전화번호 목록/전화번호 목록.py
import sys
input = sys.stdin.readline

t = int(input().strip())
answer = []

for _ in range(t):
    n = int(input().strip())
    phone_numbers = sorted(input().strip() for _ in range(n))
    
    flag = True
    for i in range(n - 1):
        if phone_numbers[i + 1].startswith(phone_numbers[i]):
            flag = False
            break

    if flag:
        answer.append("YES")
    else:
        answer.append("NO")

print("\n".join(answer))

File: 백준/Gold/5052. 전화번호 목록/전화번호 목록.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")

const t = Number(input[0])
let index = 1
const answer = []
for (let tc = 0; tc < t; tc++) {
  const n = Number(input[index++])
  const phoneNumbers = input.slice(index, index + n).sort()
  index += n
  let flag = true

  for (let i = 0; i < n - 1; i++) {
    if (phoneNumbers[i + 1].startsWith(phoneNumbers[i])) {
      flag = false
      break
    }
  }
  if (flag) {
    answer.push("YES")
  } else {
    answer.push("NO")
  }
}
console.log(answer.join("\n"))
